### ISSUE TITLE
Vim doesn't reflects colorscheme when it boot.

### ISSUE THREAD

**Issue by cmplstofB** [vim/vim#1704]
Happy Vimming, sir.
I have a problem that the acquired colorscheme isn't reflected on Vim at boot.

# Profile of Vim and so on

* OS : Debian GNU/Linux
* Virtual Terminal : Urxvt (Daemonized)
* Vim
*I compiled Vim on an Japanese environment. So this is not original text.*

````
$ vim --version
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled May  3 2017 13:37:11)
適用済パッチ(←mean “Patch”): 1-596
Compiled by <myfost>
Huge 版(←mean “Version”) without GUI.
````

# What Happen
In `$MYVIMRC` (`~/.vimrc`), I wright only:

````
let skip_defaults_vim=0
syntax on
colorscheme molokai
````

`molokai.vim` (colorscheme file) is in `~/.vim/colors` and when I type `:echo g:colors_name` in Vim, it returns “molokai.”
But, in this time, the appearance of Vim is default colorscheme, instead of molokai one.
And when I type `:source ~/.vimrc` and/or `:colorscheme molokai`, it fixed.

When I change `~/.vimrc` to this;

````
syntax on
colorscheme molokai
````

nothing changes.

===

Please give me any solution.

**Comment by nuko8** [comment]
I cannot reproduce the said problem.  Can you tell us what your vim returns with `:echo &term`?

**Comment by cmplstofB** [comment]
Thank you for reply.
Vim says `rxvt-unicode` for `:echo $TERM`, but does not say anything for `:echo $term`.

**Comment by nuko8** [comment]
Then, on your terminal, start up `urxvt` with
```
$ TERM=rxvt-unicode-256color urxvt
```
Next, on this newly opened terminal, do
```
$ vim -u NONE -c 'syntax on | colorscheme molokai'
```
Does this make a difference to you?

**Comment by cmplstofB** [comment]
Thank you, sir. But there are no difference.
I tried typing;
````
$ vim -u NONE -c 'syntax on | colorscheme molokai'
````
but the vim-booting-message[*1]'s colorscheme is default one.
I also typing;
````
$ vim -u NONE -c 'syntax on | colorscheme molokai' tmp.html
````
on terminal. File `tmp.html` is colored, but it's colorscheme isn't molokai but default one. Vim say “molokai” to `:echo g:colors_name`.

*1 Message about version and Uganda

**Comment by nuko8** [comment]
Strange.  What `:echo $TERM &term` returns after you just do `vim -u NONE -c 'syntax on | colorscheme molokai'` ?

**Comment by cmplstofB** [comment]
Vim says “rxvt-unicode rxvt-unicode.” ...what?!

Then I tried change value of `$TERM` “rxvt-unicode-256color” in different way. And succeeded.
Now I get colorscheme that based molokai!
Thank you very much and sorry for take your time.

**Comment by nuko8** [comment]
Okay, now we just ruled out the possibility that, on `urxvt`, Vim selects the default color scheme in spite of it being told to use molokai at startup, and that, nevertheless, Vim can switch the color scheme from default to molokai if the mentioned $MYVIMRC is sourced again at runtime or if molokai is explicitly set with `:colorscheme` at runtime.

All those were due to a wrong `$TERM` value for the 256-color capable `urxvt`.  Will you close this issue yourself?
